### PR TITLE
chore(deps): upgrade abseil to 20210324.1

### PR DESCRIPTION
**Pull request by devjgm**
Fixes: #6627

This version of Abseil adds proper pkgconfig support, so we no longer
need dynamically compute the Abseil `.so` files to link with, instead we
can just list the pkg-config deps that we "require" from Abseil. Indeed,
our old way of computing the libraries no longer works, so this change
was required in the same PR as the Abseil upgrade.

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/6628)

### CODE REVIEW

**Comment by devbww** [comment]
```
ld: illegal thread local variable reference to regular symbol __ZN4absl12lts_2021032413base_internal19thread_identity_ptrE for architecture x86_64
```
from `kokoro/macos/bazel`.

File: ci/generate-markdown/generate-packaging.sh [hunk 1]
```diff
@@ -199,7 +199,7 @@ cc_binary(
 
 | Library | Minimum version | Description |
 | ------- | --------------: | ----------- |
-| [Abseil][abseil-gh] | 20200923, Patch 3 | Abseil C++ common library |
```
Comment (coryan) [inline ci/generate-markdown/generate-packaging.sh:202]: We have at least one build that tests with 20200923.  I think we should not drop support for the older version, if we do, we likely have to drop support for older gRPC versions too.

Comment (devjgm) [reply ci/generate-markdown/generate-packaging.sh:202]: The issue is that this version of Abseil renames their old `.so` files and adds proper `pkg-config` support. So ideally we should start using their pkg-config files, as I've done in this version of the PR, but doing that makes it so our `Makefile` builds will no longer work w/ previous releases of Abseil.

We could also not using Abseil's `pkg-config` files that are added in this new release. But doing that would require that we change our CMake function to understand Abseil's new `.so` naming.

Comment (coryan) [reply ci/generate-markdown/generate-packaging.sh:202]: I think I follow how we end up here.

I am trying to think about this from the perspective of customers reading this document, maybe for the first time.  They read `Abseil >= 20210324` and also read `gRPC >= 1.35` which supposedly works with (requires?) `Abseil == 20200923`.  That seems complicated.  Which one is it?

Anyway, feel free to ignore this.  The thing that @devbww points out seems more important.

**Comment by devjgm** [comment]
> ```
> ld: illegal thread local variable reference to regular symbol __ZN4absl12lts_2021032413base_internal19thread_identity_ptrE for architecture x86_64
> ```
> 
> from `kokoro/macos/bazel`.

It looks like an Abseil issue. I filed https://github.com/abseil/abseil-cpp/issues/965

**Comment by devjgm** [comment]
Closing this PR for now. Abseil is working up a fix for `ld: illegal thread local variable reference to regular symbol __ZN4absl12lts_2021032413base_internal19thread_identity_ptrE for architecture x86_64` and they'll cut another patch release.

**Pull request closed without merging**